include/dxc/WinAdapter.h

```diff
@@ -748,6 +748,13 @@ template <class T> class CComPtr : public CComPtrBase<T> {
     return *this;
   }
 
+  // NOTE: This conversion constructor is not part of the official CComPtr spec;
+  // however, it is needed to convert CComPtr<Q> to CComPtr<T> where T derives from Q on Clang.
+  // MSVC compiles this conversion as first a call to CComPtr<Q>::operator T*, followed by CComPtr<T>(T*),
```

Comment:
I wonder if this has something to do with MSVC's two-phase lookup. I say this because to be able to build DXC using Clang-CL, I had to pass in `/Zc:twoPhase-`  (see [here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/dawn/third_party/gn/dxc/BUILD.gn;l=473?q=dxc%20BUILD.gn&ss=chromium)).